apps/ui/components/reservation-unit/Head.tsx

```diff
@@ -107,6 +107,40 @@ const UnitName = styled(H3).attrs({ as: "h2" })`
   }
 `;
 
+const NonReservableNotification = ({
+  reservationUnit,
+}: {
+  reservationUnit: ReservationUnitByPkType;
+}) => {
+  const { t } = useTranslation();
+  const getFutureText = () =>
+    isReservationStartInFuture(reservationUnit)
+      ? t("reservationUnit:notifications.futureOpening", {
+          date: reservationUnit.reservationBegins
+            ? formatDate(reservationUnit.reservationBegins, "d.M.yyyy")
+            : "",
+          time: reservationUnit.reservationBegins
+            ? formatDate(reservationUnit.reservationBegins, "H.mm")
+            : "",
+        })
+      : false;
+  const returnText = t(
+    `reservationUnit:notifications.${
+      reservationUnit.reservationKind ===
+      ReservationUnitsReservationUnitReservationKindChoices.Season
+        ? "onlyRecurring"
+        : getFutureText() || "notReservable"
```

Comment:
Is the future text supposed to be inside the translation?

Reply:
Nope. Refactored the whole thing to a more readable form.